robotpy_build/generator_data.py

```diff
@@ -27,14 +23,14 @@ def __init__(self, data: HooksDataYaml):
         self.data = data
 
         # report data
-        self.functions = {}
-        self.classes = {}
-        self.enums = {}
-        self.attributes = {}
+        self.functions: Dict[str, bool] = {}
+        self.classes: Dict[str, Dict] = {}
+        self.enums: Dict[str, bool] = {}
+        self.attributes: Dict[str, bool] = {}
 
     def get_class_data(self, name: str) -> ClassData:
-        data = self.data.classes.get(name, _missing)
```

Comment:
The data could legitimately be None, and we don't want to flag that.

Reply:
Don't you change all Nones to empty instances in the validators?

Reply:
Yes, we always do that. However, I believe some of these allow None to be 'not missing' because yaml of the form:

```
foo:
```

Ends up being `{"foo": None}`, which should not be 'missing'.

Reply:
Sure, which becomes `{"foo": ClassData()}` (for example) after validation, right?

Reply:
Correct.

Reply:
... so it is impossible to get here with `{"foo": None}`.